### ISSUE TITLE
GET params not passed on properly with forward

### ISSUE THREAD

**Issue by b10m** [PerlDancer/Dancer2#1245]
Considering an app like this:
```
get '/foo' => sub {
    return "Got: ".query_parameters->get('bar');
};

get '/foo/bar' => sub {
    forward '/foo', { bar => 'baz' };
};
```
You'd expect to see "Got: baz" when the /foo/bar URL was triggered, yet the `bar` parameter doesn't get added to the query_parameters (although the docs do suggest this is the preferred way of both adding params and reading params).

Using param() rather than query_parameters() works for now, but the additional param should simply be added to the query_parameters multivalue'd hash, IMHO.

**Comment by xsawyerx** [comment]
I remember seeing something like that. I don't remember if we opened a ticket, fixed it in a branch, or what.

**Comment by xsawyerx** [comment]
Ah, here we go. Related issue: #1116.

**Comment by veryrusty** [comment]
If we consider `forward '/foo', { bar => 'baz' };` as being the same as `forward '/foo?bar=baz';`, that is treat those params as query params;  then we can make `request->query_parameters` and `request->params('query')` be consistent and include those new params.

**Comment by b10m** [comment]
... and of course the same for `forward '/foo', { bar => 'baz' }, { method => 'POST' }` and `body_parameters`.

**Comment by b10m** [comment]
FWIW, the [documentation](https://metacpan.org/pod/Dancer2::Manual#forward) suggests it does this (emphasis added):

> More formally; when forward is executed, the current dispatch of the route is aborted, the request is modified (**altering query params** or request method), and the modified request following a new route is dispatched again.

**Comment by veryrusty** [comment]
I've started on resolving this (and #1116).  Commits are in the branch  `bugfix/hmv_forward`.

Any new params supplied to the `forward` get added to `params`, `query_params` and `query_parameters`, which should resolve this issue. (Yes, even if the method is a POST, the new params are *query* params).

For 1116; the HMV for plack body params from the current request is cloned into the new request.

No PR; as I haven't added test cases. Feel free to pick up that work and add tests :)

**Comment by cromedome** [comment]
Resolved by 6dc4a48d5ef4a11cc0ff6a3defbc3f32acb23c73. Closing. Thanks @veryrusty!
